lib/location/scan_cellular.c

```diff
@@ -24,11 +24,28 @@ static struct lte_lc_cells_info scan_cellular_info = {
 };
 
 static bool running;
+static bool timeout_occurred;
```

Comment:
Not sure if it makes any difference, but this could be `volatile`. It's being set and checked from different threads.